File: backend/src/main/java/com/bartoszkorec/warehouse_management/dto/response/OrderResponse.java
package com.bartoszkorec.warehouse_management.dto.response;

import com.bartoszkorec.warehouse_management.dto.GridDto;
import com.bartoszkorec.warehouse_management.model.Point;

import java.util.List;

public record OrderResponse(List<GridDto> warehouses, List<OrderLocationDto> locations) {
    public record OrderLocationDto(String label, Point point) {
    }
}

File: frontend/src/app/models/order-response.ts
export interface OrderResponse {
  warehouses: GridDto[];
  locations: OrderLocationDto[];
}

export interface GridDto {
  id: number;
  layout: number[][];
}

export interface OrderLocationDto {
  label: string;
  point: PointDto;
}

export interface PointDto {
  gridIndex: number;
  x: number;
  y: number;
}
